packages/cli/src/modules/insights/__tests__/insights-pruning.service.test.ts

```diff
@@ -0,0 +1,169 @@
+} from '../database/entities/__tests__/db-utils';
+import { InsightsByPeriodRepository } from '../database/repositories/insights-by-period.repository';
+import { InsightsPruningService } from '../insights-pruning.service';
+import { InsightsConfig } from '../insights.config';
+
+// Initialize DB once for all tests
+beforeAll(async () => {
+	await testDb.init();
+});
+
+describe('InsightsPrunningService', () => {
+	let insightsConfig: InsightsConfig;
+	let insightsByPeriodRepository: InsightsByPeriodRepository;
+	let insightsPruningService: InsightsPruningService;
+	beforeAll(async () => {
+		insightsConfig = Container.get(InsightsConfig);
+		insightsConfig.maxAgeDays = 10;
+		insightsConfig.pruneCheckIntervalHours = 1;
+		insightsPruningService = Container.get(InsightsPruningService);
+		insightsByPeriodRepository = Container.get(InsightsByPeriodRepository);
+	});
+
+	test('old insights get pruned successfully', async () => {
+		// ARRANGE
+		const project = await createTeamProject();
+		const workflow = await createWorkflow({}, project);
+
+		await createMetadata(workflow);
+
+		const timestamp = DateTime.utc().minus({ days: insightsConfig.maxAgeDays + 1 });
+		await createCompactedInsightsEvent(workflow, {
+			type: 'success',
+			value: 1,
+			periodUnit: 'day',
+			periodStart: timestamp,
+		});
+
+		// ACT
+		await insightsPruningService.pruneInsights();
+
+		// ASSERT
+		expect(await insightsByPeriodRepository.count()).toBe(0);
+	});
+
+	test('insights newer than maxAgeDays do not get pruned', async () => {
+		// ARRANGE
+		const project = await createTeamProject();
+		const workflow = await createWorkflow({}, project);
+
+		await createMetadata(workflow);
+
+		const timestamp = DateTime.utc().minus({ days: insightsConfig.maxAgeDays - 1 });
+		await createCompactedInsightsEvent(workflow, {
+			type: 'success',
+			value: 1,
+			periodUnit: 'day',
+			periodStart: timestamp,
+		});
+
+		// ACT
+		await insightsPruningService.pruneInsights();
+
+		// ASSERT
+		expect(await insightsByPeriodRepository.count()).toBe(1);
+	});
+
+	test('pruning timer is scheduled on start and rescheduled after each run', async () => {
+		jest.useFakeTimers();
+		const insightsByPeriodRepository = mock<InsightsByPeriodRepository>({
+			pruneOldData: async () => {
+				return { affected: 0 };
+			},
+		});
+		const insightsPruningService = new InsightsPruningService(
+			insightsByPeriodRepository,
+			insightsConfig,
+			mockLogger(),
+		);
+		const pruneSpy = jest.spyOn(insightsPruningService, 'pruneInsights');
+		const scheduleNextPruneSpy = jest.spyOn(insightsPruningService as any, 'scheduleNextPrune');
+
+		try {
+			insightsPruningService.startPruningTimer();
+
+			// Wait for pruning timer promise to resolve
+			await jest.advanceTimersToNextTimerAsync();
+
+			expect(pruneSpy).toHaveBeenCalledTimes(1);
+			expect(scheduleNextPruneSpy).toHaveBeenCalledTimes(2);
+		} finally {
+			jest.useRealTimers();
+			insightsPruningService.stopPruningTimer();
+		}
```

Comment:
Should this `finally` block be in a `beforeEach` or `afterEach` instead? If so, same elsewhere.